game/scripts/vscripts/Sections/Chapter6/SectionOpening.ts

```diff
@@ -0,0 +1,280 @@
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast ss
+            castOnPosition(dummyLocation, invokerName, 3, 0.3),
+
+            // prepare meatball (2,2,1)
+            castNoTarget(invokerName, 1),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast meatball
+            castOnPosition(dummyLocation, invokerName, 3, 0.3),
+
+            // prepare blast (2,1,0)
+            castNoTarget(invokerName, 0),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast blast
+            castOnPosition(dummyLocation, invokerName, 3, 0.3),
+
+            // prepare cold snap (0,0,0)
+            castNoTarget(invokerName, 0),
+            castNoTarget(invokerName, 0),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast cold snap
+            castOnTarget(axeName, invokerName, 3, 0.3),
+
+            // prepare ice wall (0,0,2)
+            castNoTarget(invokerName, 2),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast ice wall
+            castOnPosition(dummyLocation, invokerName, 3, 0.3),
+
+            // prepare summons (0,2,2)
+            castNoTarget(invokerName, 2),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast summons
+            castNoTarget(invokerName, 3, 0.3),
+
+            // prepare alacrity (1,1,2)
+            castNoTarget(invokerName, 1),
+            castNoTarget(invokerName, 1),
+            castNoTarget(invokerName, 5, 0.5),
+
+            // cast alacrity
+            tg.immediate(context => context[invokerName].SetAttackCapability(UnitAttackCapability.RANGED_ATTACK)),
+            castOnTarget(invokerName, invokerName, 3, 0.3),
+
+            tg.wait(3),
+            tg.immediate(context => context[invokerName].RemoveSelf()),
+            tg.immediate(_ => Entities.FindAllByClassname("npc_dota_invoker_forged_spirit").forEach(forgedSpirit => forgedSpirit.RemoveSelf())),
+        ]),
+        tg.immediate(_ => goalWatchInvoker.complete()),
+
+        tg.audioDialog(LocalizationKey.Script_6_Opening_6, LocalizationKey.Script_6_Opening_6, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+        tg.audioDialog(LocalizationKey.Script_6_Opening_7, LocalizationKey.Script_6_Opening_7, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+        tg.audioDialog(LocalizationKey.Script_6_Opening_8, LocalizationKey.Script_6_Opening_8, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+        tg.audioDialog(LocalizationKey.Script_6_Opening_9, LocalizationKey.Script_6_Opening_9, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+        tg.wait(3),
+    ]))
+
+    graph.start(GameRules.Addon.context, () => {
+        print("Completed", sectionName);
+        complete();
+    });
+}
+
+function onStop() {
+    print("Stopping", sectionName);
+
+    if (graph) {
+        graph.stop(GameRules.Addon.context);
+        graph = undefined;
+    }
+}
+
+function castNoTarget(casterName: string, index: number, wait = 0.1) {
```

Comment:
Those functions are useful and can probably be put as steps. I know we're technically done, but for the other people who might want follow our tutorial methods. Up to you though.

Reply:
Ye maybe we can replace all of our cast stuff in another PR

Reply:
"casting all his balls" a sentence I never saw coming lol